### ISSUE TITLE
Cannot install Helm chart : 400 bad request

### ISSUE THREAD

**Issue by antoineozenne-at-leocare** [Azure/AppConfiguration#809]
The installation of the Helm chart with the following command fails:

```sh
helm install azureappconfiguration.kubernetesprovider \
     oci://mcr.microsoft.com/azure-app-configuration/helmchart/kubernetes-provider \
     --version 1.0.0-preview4 \
     --namespace azappconfig-system \
     --create-namespace
```

It returns:

```console
Error: INSTALLATION FAILED: failed to copy: httpReadSeeker: failed open: unexpected status code https://mcr.microsoft.com/v2/azure-app-configuration/helmchart/kubernetes-provider/blobs/sha256:a486cb57298a28209dbb4e1c416efb648f52f1a24f99d7dac798181f6f303dd3: 400 Bad Request
```

**Comment by RichardChen820** [comment]
I can't reproduce it on my end, can you give it another try to check whether it's a transient MCR problem?

**Comment by antoineozenne-at-leocare** [comment]
It may be a regression.

It doesn't work with the following Helm version:

```console
version.BuildInfo{Version:"v3.13.0", GitCommit:"825e86f6a7a38cef1112bfa606e4127a706749b1", GitTreeState:"clean", GoVersion:"go1.21.1"}
```

However, it works with the last stable Helm release:

```console
version.BuildInfo{Version:"v3.12.3", GitCommit:"3a31588ad33fe3b89af5a2a54ee1d25bfe6eaa5e", GitTreeState:"clean", GoVersion:"go1.20.7"}
```

**Comment by antoineozenne-at-leocare** [comment]
Should we consider that it is a Helm bug? Or something else?

**Comment by RichardChen820** [comment]
Not quite sure yet, I'm not sure what protocol standard the helm client and MCR are following.  Apparently, there's some mismatch between them now.  I'm reaching MCR guys to see if I can get more information about this, I will get back to you if I get anything from them, please stay tuned

**Comment by RichardChen820** [comment]
Tracking issue https://github.com/helm/helm/issues/12463

**Comment by RichardChen820** [comment]
It's a helm regression, should be fixed in v3.13.1 https://github.com/helm/helm/pull/12424
